scipy/cluster/hierarchy.py

```diff
@@ -1481,9 +1481,10 @@ def to_tree(Z, rd=False):
             raise ValueError(('Corrupt matrix Z. Index to derivative cluster '
                               'is used before it is formed. See row %d, '
                               'column 1') % fj)
-        nd = ClusterNode(i + n, d[fi], d[fj], Z[i, 2])
-        #                 ^ id   ^ left ^ right ^ dist
-        if Z[i, 3] != nd.count:
+        
+        nd = ClusterNode(i + n, d[fi], d[fj], row[2])
+        #                ^ id   ^ left ^ right ^ dist
+        if row[3] != nd.count:
```

Comment:
I suppose it is a bit more Pythonic, though perhaps debatable on whether it is worth the churn.

I don't see any issues and CI is green so that's good I suppose. Note that codecov shows that some of the error handling is not covered by tests at the moment, but does seems fairly harmless.